Inspect validation

diaabadr opened the pull request:
@viferga 
I have added the new updates on the master branch with the  `ensureDeploymentsExist`

viferga commented:
I don't know why but many tests are failing, you should review them, otherwise we cannot merge this. I'm setting this PR as a draft until it passes the tests.

diaabadr commented:
@viferga 
I think they are failing because of the secrets. workflows on a pull requests doesn't have access to the secrets. when i run it on my forked repo it passes the tests that doesn't require plans such as login, inspect, etc. 

![image](https://user-images.githubusercontent.com/77173710/235283042-61f30ed8-0ed5-4c29-ba09-8b93155892b4.png)

It sees the secrets in my forked repo

![image](https://user-images.githubusercontent.com/77173710/235283114-bf2df608-0af1-42d4-b721-44e1c6597721.png)

but it doesn't see them on the pr workflow

![image](https://user-images.githubusercontent.com/77173710/235283140-9fd7023c-02d9-4bd7-a1a4-2f42c308c272.png)

I think that's the problem. what do you see?

diaabadr commented:
I have added extra validation at the login with email and password flags. if the email and password are empty. the CLI asks for login with the `auth choices` (token or email). it doesn't consider them as invalid credentials so it doesn't throw an error to catch it and fail the login test. so i added a check on the info messages if it includes `login successful` or not.

viferga commented:
@diaabadr I re-run the tests by doing a small change in the code, and they failed. You can review now and solve the issues if you want, meanwhile I am going to leave this PR as a draft.
